modules/dublincore/src/main/java/org/opencastproject/metadata/dublincore/MetadataField.java

```diff
@@ -1041,60 +1038,84 @@ public static JObject mapToJSON(Map<String, String> map) {
     return obj(fields);
   }
 
-  /**
-   * A convenience function for converting properties values into their java equivalents
-   *
-   * @param key
-   *          The key for the metadata property.
-   * @param value
-   *          The value for the metadata property.
-   */
-  public void setValue(String key, String value) {
-    switch (key) {
-      case CONFIG_COLLECTION_ID_KEY:
-        this.collectionID = Opt.some(value);
-        break;
-      case CONFIG_PATTERN_KEY:
-        this.pattern = Opt.some(value);
-        break;
-      case CONFIG_DELIMITER_KEY:
-        this.delimiter = Opt.some(value);
-        break;
-      case CONFIG_INPUT_ID_KEY:
-        this.inputID = value;
-        break;
-      case CONFIG_LABEL_KEY:
-        this.label = value;
-        break;
-      case CONFIG_LIST_PROVIDER_KEY:
-        this.listprovider = Opt.some(value);
-        break;
-      case CONFIG_NAMESPACE_KEY:
-        this.namespace = Opt.some(value);
-        break;
-      case CONFIG_ORDER_KEY:
-        try {
-          Integer orderValue = Integer.parseInt(value);
-          this.order = Opt.some(orderValue);
-        } catch (NumberFormatException e) {
-          logger.warn("Unable to parse order value {} of metadata field {}", value, this.getInputID(), e);
-          this.order = Opt.none();
-        }
-        break;
-      case CONFIG_OUTPUT_ID_KEY:
-        this.outputID = Opt.some(value);
-        break;
-      case CONFIG_READ_ONLY_KEY:
-        this.readOnly = Boolean.valueOf(value);
-        break;
-      case CONFIG_REQUIRED_KEY:
-        this.required = Boolean.valueOf(value);
-        break;
-      case CONFIG_TYPE_KEY:
-        this.type = Type.valueOf(value.toUpperCase());
-        break;
+  public static MetadataField createMetadataField(Map<String,String> dublinCoreProperties) {
+    Opt<String> collectionID = dublinCoreProperties.containsKey(CONFIG_COLLECTION_ID_KEY)
+            ? Opt.some(dublinCoreProperties.get(CONFIG_COLLECTION_ID_KEY)) : Opt.none();
+    String pattern = dublinCoreProperties.containsKey(CONFIG_PATTERN_KEY)
+            ? dublinCoreProperties.get(CONFIG_PATTERN_KEY) : "yyyy-MM-dd'T'HH:mm:ss.SSS'Z'";
```

Comment:
Maybe add this default value to the documentation of the metadata field _pattern_ on https://docs.opencast.org/develop/admin/configuration/metadata/

Reply:
I'm not at all sure whether that is consistent with other parts of the code, so I'm gonna refrain from that for now until all of this is a bit more sane...